sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+
+[How to efficiently upload large amount of small blobs?](#how-to-efficiently-upload-large-amount-of-small-blobs)
+
+[How to ensure data integrity with transactional checksum?](#how-to-ensure-data-integrity-with-transactional-checksum)
+
+
+**Design and Concepts**
+
+[Thread safety guarantees](#thread-safety-guarantees)
+
+[Http pipeline and policies](#http-pipeline-and-policies)
+
+[Will the SDK retry failed requests? What is the default retry logic? How can I customize retry behavior?](#will-the-sdk-retry-failed-requests-what-is-the-default-retry-logic-how-can-i-customize-retry-behavior)
+
+
+**Troubleshooting**
+
+[Why do I see 503 Server Busy errors? What should I do in this case?](#why-do-i-see-503-server-busy-errors-what-should-i-do-in-this-case)
+
+[How to troubleshoot 403 errors?](#how-to-troubleshoot-403-errors)
+
+## How to use list-operations? Why doesn't list-operation return all blobs in a container?
+
+C++ SDK applies a different design for pageable operations from other languages.
+It involves a nested-loop to iterate over all items in a container.
+The outer loop does network I/O operations.
+The inner loop gets called for every paged result and doesn't do I/O.
+
+Below is an example of listing all blobs in a blob container.
+
+```C++
+for (auto page = blobContainerClient.ListBlobs(); page.HasPage(); page.MoveToNextPage()) {
+  for (auto& blob : page.Blobs) {
+    std::cout << blob.Name << std::endl;
+  }
+}
+```
+
+Sometimes a paged result may contain multiple collections and you may want to iterate over all of them.
+
+```C++
+for (auto page = directoryClient.ListFilesAndDirectories(); page.HasPage(); page.MoveToNextPage())
+{
+  for (const auto& d : page.Directories)
+  {
+    std::cout << "Directory: " << d.Name << std::endl;
+  }
+  for (const auto& f : page.Files)
+  {
+    std::cout << "File: " << f.Name << std::endl;
+  }
+}
+```
+
+## Can I specify an API-version other than the default one?
+
+Yes, each client options takes an `ApiVersion` as optional parameter, with which you can specify an API-version used for all the HTTP requests from this client.
+Client spawned from another client instance will inherit the settings.
+
+```C++
+// serviceClient sends HTTP requests with default API-version, which will change as version evolves.
+auto serviceClient = BlobServiceClient::CreateFromConnectionString(GetConnectionString());
+
+// serviceClient sends HTTP requests with specified API-version.
+BlobClientOptions clientOptions;
+clientOptions.ApiVersion = "2019-12-12";
+auto serviceClient = BlobServiceClient::CreateFromConnectionString(GetConnectionString, clientOptions);
+// containerClient inherits this setting.
+auto containerClient = serviceClient.GetBlobContainerClient(containerName);
+```
+
+Please note that multiple-version is not supported yet by Storage C++ SDK (This may change in the future).
```

Comment:
Stating `yet` makes it seem like we already have plans to do so, but just haven't.
```suggestion
Please note that multiple-version is not supported by the Storage C++ SDK (this may change in the future).
```